core/src/main/java/org/jdbi/v3/BeanMapper.java

```diff
@@ -143,44 +88,58 @@ public T map(int row, ResultSet rs, StatementContext ctx)
         ResultSetMetaData metadata = rs.getMetaData();
 
         for (int i = 1; i <= metadata.getColumnCount(); ++i) {
-            String name = metadata.getColumnLabel(i).toLowerCase();
+            String name = metadata.getColumnLabel(i);
 
-            PropertyDescriptor descriptor = properties.get(name);
+            final Optional<PropertyDescriptor> maybeDescriptor =
+                    descriptorByColumnCache.computeIfAbsent(name, this::descriptorForColumn);
 
-            if (descriptor != null) {
-                Class type = descriptor.getPropertyType();
+            if (!maybeDescriptor.isPresent()) {
```

Comment:
Only reason I did it this way is because I rather like escaping the loop early, rather than wrapping all of the logic in an `if` block... fewer levels of useless indentation :)